envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/utils/ReactorUtilsTest.kt

```diff
@@ -73,6 +73,33 @@ class ReactorUtilsTest {
         assertThat(buffered).isEqualTo(2.0)
     }
 
+    @Test
+    fun `should measure discarded items of backpressureLatest`() {
```

Comment:
Have you tried using [StepVerifier](https://projectreactor.io/docs/test/release/api/index.html?reactor/test/StepVerifier.html) in tests? I wonder if that's something wrong with it since you're using your own methods, like using `CountDownLatch` in `subscribeRequestingN`.

Reply:
Thanks, I didn't know about that. After a quick look it looks like this is the right tool. I will use it next time.